### PR TITLE
python312Packages.jianpu-ly: init at 1.801

### PR DESCRIPTION
**Pull request by ifurther**
## Description of changes

Add jianpu-ly application

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/jianpu-ly/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  lilypond,
+}:
+
+buildPythonPackage rec {
+  pname = "jianpu-ly";
+  version = "1.801";
+  format = "setuptools";
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/jianpu-ly/default.nix:11]: - #253154
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/jianpu-ly/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  src = fetchPypi {
+    inherit version;
+    pname = "jianpu_ly";
+    sha256 = "sha256-piK9Ym94cKdaHGz/ogX7ylyAF1ww0jCdRXnHN6lu2MI=";
+  };
+
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/jianpu-ly/default.nix:18]:
```suggestion

  build-system = [ setuptools ];

```

File: pkgs/development/python-modules/jianpu-ly/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+buildPythonPackage rec {
+  pname = "jianpu-ly";
+  version = "1.801";
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "jianpu_ly";
+    sha256 = "sha256-piK9Ym94cKdaHGz/ogX7ylyAF1ww0jCdRXnHN6lu2MI=";
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/jianpu-ly/default.nix:16]: - #325892

```suggestion
    hash = "sha256-piK9Ym94cKdaHGz/ogX7ylyAF1ww0jCdRXnHN6lu2MI=";
```

File: pkgs/development/python-modules/jianpu-ly/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  nativeBuildInputs = [ lilypond ];
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/jianpu-ly/default.nix:19]:
```suggestion
  dependencies = [ lilypond ];
```

File: pkgs/development/python-modules/jianpu-ly/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+  pythonImportsCheck = [ "jianpu_ly" ];
+  # no tests in shipped with upstream
+  doCheck = false;
+
+  meta = with lib; {
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/jianpu-ly/default.nix:25]: - #208242
```suggestion
  meta = {
```

File: pkgs/development/python-modules/jianpu-ly/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    homepage = "https://ssb22.user.srcf.net/mwrhome/jianpu-ly.html";
+    description = "jianpu-ly is a Python program that assists with printing jianpu";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ifurther ];
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/jianpu-ly/default.nix:29]:
```suggestion
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ ifurther ];
```

File: pkgs/development/python-modules/jianpu-ly/default.nix [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    platforms = platforms.all;
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/jianpu-ly/default.nix:30]: Uneeded for `buildPythonPackage`
```suggestion
```

File: pkgs/development/python-modules/jianpu-ly/default.nix [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchPypi {
+    inherit version;
+    pname = "jianpu_ly";
+    sha256 = "sha256-piK9Ym94cKdaHGz/ogX7ylyAF1ww0jCdRXnHN6lu2MI=";
+  };
+
+  nativeBuildInputs = [ lilypond ];
+
+  pythonImportsCheck = [ "jianpu_ly" ];
+  # no tests in shipped with upstream
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/jianpu-ly/default.nix:22]: nit
```suggestion
  pythonImportsCheck = [ "jianpu_ly" ];

  # no tests in shipped with upstream
```

File: pkgs/development/python-modules/jianpu-ly/default.nix [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://ssb22.user.srcf.net/mwrhome/jianpu-ly.html";
+    description = "jianpu-ly is a Python program that assists with printing jianpu";
```
Comment (Sigmanificient) [inline pkgs/development/python-modules/jianpu-ly/default.nix:27]: *`meta.description` must: [...] Not start with the package name. *
> https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

```suggestion
    description = "Assists with printing jianpu";
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/check-manifest/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/check-manifest/default.nix:1]: This got already fixed in #325567 Please rebase your branch on master

File: pkgs/development/python-modules/scalene/default.nix [hunk 10]
```diff
@@ -37,9 +39,13 @@ buildPythonPackage rec {
...
     psutil
-    pynvml
     rich
-  ];
+    pynvml
+  ] ++ lib.optionals stdenv.hostPlatform.isLinux [ nvidia-ml-py ];
+
+  pythonRemoveDeps = [
+    "nvidia-ml-py3"
+  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [ "nvidia-ml-py" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/scalene/default.nix:48]:
```suggestion
  ] ++ lib.optionals (!stdenv.hostPlatform.isLinux) [ "nvidia-ml-py" ];

```

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**